### PR TITLE
0.2/optimized properties

### PR DESCRIPTION
**Pull request by seanbarzilay**
#31

### CODE REVIEW

**Comment by rmagen** [comment]
Bulk querying comes up in most of our Step implementations. We also need a simple way to configure the different bulk sizes and behaviors (e.g. Incremental bulk size). Maybe create a helper class for all of this?
A different Or maybe have a UniBulkStep that

**Comment by seanbarzilay** [comment]
Fixed Properties strategy 
Deleted UniGraphVertexPropertiesSideEffectStep since it is no longer needed
Implemented UniBulkStep and converted steps to use it

File: unipop-core/src/org/unipop/process/vertex/UniGraphVertexStep.java [hunk 1]
```diff
@@ -76,38 +87,47 @@ public UniGraphVertexStep(VertexStep<E> vertexStep, ControllerManager controller
...
                 .<Edge>flatMap(ConversionUtils::asStream)
                 .<Traverser.Admin<E>>flatMap(edge -> toTraversers(edge, idToTraverser)).iterator();
+        if (!this.returnsVertex || (propertyKeys != null && propertyKeys.size() == 0))
+            return traversersIterator;
+        return getTraversersWithProperties(traversersIterator);
+    }
+
+    private Iterator<Traverser.Admin<E>> getTraversersWithProperties(Iterator<Traverser.Admin<E>> traversers) {
+        List<Traverser.Admin<E>> copyTraversers = ConversionUtils.asStream(traversers).collect(Collectors.toList());
+        List<DeferredVertex> deferredVertices = copyTraversers.stream().map(traverser -> ((DeferredVertex) traverser.get())).collect(Collectors.toList());
```
Comment (rmagen) [inline unipop-core/src/org/unipop/process/vertex/UniGraphVertexStep.java]: Why is this still here?

Comment (seanbarzilay) [inline unipop-core/src/org/unipop/process/vertex/UniGraphVertexStep.java]: what do you mean?

**Pull request merged**